twine/repository.py

```diff
@@ -77,12 +77,19 @@ def __init__(
 
     @staticmethod
     def _make_adapter_with_retries() -> adapters.HTTPAdapter:
-        retry = urllib3.Retry(
+        retry_kwargs = dict(
             connect=5,
             total=10,
-            method_whitelist=["GET"],
             status_forcelist=[500, 501, 502, 503],
         )
+
+        try:
+            retry = urllib3.Retry(allowed_methods=["GET"], **retry_kwargs)
+        except TypeError:  # pragma: no cover
```

Comment:
I tested this locally, via https://github.com/pypa/twine/pull/719/commits/b07f7db08b7ca4747d45752dbc036c21bb3ac460. But the `-e py36` isn't run by the GitHub Action; instead, it runs `-e py` to use the job's matrix version:

https://github.com/pypa/twine/blob/fd5719877cd382fff7bae9821878d3fd2ed0c328/.github/workflows/main.yml#L40-L52

With that in mind, this felt like a reasonable time to use `no cover`.